torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -522,14 +522,42 @@ Tensor clamp_backward(const Tensor & grad, const Tensor &self, const optional<Sc
   }
 }
 
-Tensor mm_mat1_backward(const Tensor & grad, const Tensor & mat2, const Tensor & mat1, const Scalar & alpha) {
-  // if input was column-major, return grad as column-order for efficiency
-  if (mat1.is_sparse()) {
-    throw std::runtime_error("calculating the gradient of a sparse Tensor argument to mm is not supported.");
+// This function is used by load_derivatives.py to replace tensor.strides()
+// calls that appear in derivative formulas. This function checks if the tensor
+// is strided, and if not, produces an error telling the user how to fix the
+// issue. If requires_grad is not set for the corresponding inputs of the
+// operation, no error is thrown and an empty array is returned since there
+// will be no backward pass in this case.
+//
+// Args:
+//  self              Tensor to call .strides() on
+//  self_name         Name of `input` tensor, from derivative formula
+//  grad_vars         List of tensor inputs the derivative formula applies to
+//  grad_vars_names   Comma separated string of the names of `grad_vars` tensors
+at::IntArrayRef backward_strides_or_error(const Tensor & self, c10::string_view const & self_name, at::TensorList grad_vars, c10::string_view const & grad_var_names) {
+  bool requires_grad = false;
+  for (auto grad_var : grad_vars) {
```

Comment:
This is a workaround because in an idea world, this function shouldn't even be called if none of these require grads right?
Maybe add a comment in the function description above mentioning this if it's the case.